src/services/usersLoginNotificationService.ts

```diff
@@ -0,0 +1,39 @@
+import * as t from "io-ts";
+
+import { QueueClient, QueueSendMessageResponse } from "@azure/storage-queue";
+
+import { UTCISODateFromString } from "italia-ts-commons/lib/dates";
+import { FiscalCode } from "italia-ts-commons/lib/strings";
+
+export const UserLogin = t.interface({
+  fiscalCode: FiscalCode,
+  lastLoginAt: UTCISODateFromString,
+  source: t.keyof({
+    cie: null,
+    spid: null
+  })
+});
```

Comment:
Can we use something like this? IMHO is more clear.
```suggestion
const sourceType = t.union([t.literal("spid"), t.literal("cie")]);
type sourceType = t.TypeOf<typeof sourceType>;

export const UserLogin = t.interface({
  fiscalCode: FiscalCode,
  lastLoginAt: UTCISODateFromString,
  source: sourceType
});
```

Reply:
@BurnedMarshal  You mean to extract the source type, to use "union" instead of "keyof", or both?

Reply:
"union" instead of "keyof" could be enough.

Reply:
@BurnedMarshal https://github.com/gcanti/io-ts/tree/1.x#union-of-string-literals